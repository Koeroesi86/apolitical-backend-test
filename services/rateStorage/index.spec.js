const rateStorage = require("./index");

describe('rateStorage', () => {
  beforeEach(() => {
    rateStorage._visitorCounter = {};
  });

  it('should get default value', async () => {
    expect(await rateStorage.get('127.0.0.1')).toMatchSnapshot();
  });

  it('should set value', async () => {
    await rateStorage.set('127.0.0.1', 100)
    expect(rateStorage._visitorCounter).toMatchSnapshot();
  });

  it('should retrieve set value', async () => {
    await rateStorage.set('127.0.0.1', 100)
    expect(await rateStorage.get('127.0.0.1')).toMatchSnapshot();
  });
})