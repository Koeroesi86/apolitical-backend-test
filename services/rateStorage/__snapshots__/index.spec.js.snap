// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rateStorage should get default value 1`] = `0`;

exports[`rateStorage should retrieve set value 1`] = `100`;

exports[`rateStorage should set value 1`] = `
Object {
  "127.0.0.1": 100,
}
`;
